### PR TITLE
More filepicker improvements

### PR DESCRIPTION
**Pull request by icewind1991**
- Allow selecting the root folder when picking a folder
- Allow selecting a folder after using the breadcrumb when picking a folder
- Use the new DAV based api to get the file list instead of the old ajax endpoint
- Show the filesize in the file picker

### CODE REVIEW

**Comment by nickvergessen** [comment]
Would also be nice, if double clicking a valid entry would "choose" it.

**Comment by jancborchardt** [comment]
@icewind1991 nice work! Added on top of it some nice layout fixes:
capture du 2016-10-18 11-42-50

Please review @nextcloud/javascript @nextcloud/designers :)

**Comment by icewind1991** [comment]
Fixed an issue with the "file exists" dialog that was introduced by @jancborchardt's changes

File: core/js/oc-dialogs.js [hunk 1]
```diff
@@ -741,50 +746,59 @@ var OCdialogs = {
...
-			$.each(response.data.files, function(index, file) {
-				if (file.type === 'dir') {
-					dirs.push(file);
+		var filter = this.$filePicker.data('mimetype');
+		if (typeof(filter) === "string") {
+			filter = [filter];
+		}
+		self.filepicker.filesClient.getFolderContents(dir).then(function(status, files) {
+			files = files.filter(function(file) {
+				return file.type === 'dir' || filter.indexOf(file.mimetype) !== -1;
```
Comment (MorrisJobke) [inline core/js/oc-dialogs.js]: By default the mimetype filter is `undefined, which causes an error here

**Review by MorrisJobke — CHANGES_REQUESTED** [review]
Calling `OCdialogs.filepicker('test')`doesn't work properly anymore - only folders are shown and choosing doesn't work anymore :(

**Comment by jancborchardt** [comment]
@icewind1991 I added proper left margin for the title. But the filepicker doesn’t seem to work anymore:
`oc-dialogs.js?v=a057f20…:799 Uncaught TypeError: Cannot read property 'find' of undefined`
capture du 2016-10-19 14-42-28
cc @nextcloud/javascript

**Comment by icewind1991** [comment]
@jancborchardt can't reproduce your issue

**Comment by jancborchardt** [comment]
Ok, then I guess it’s ready for @nextcloud/javascript for review :)

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
I fixed an overflow issue and applied the proper padding on the right side (it was changed from 10 to 15 px).

I tested this with IE11, Edge, Safari, Firefox and Chrome and it looks good 👍

**Comment by jancborchardt** [comment]
@MorrisJobke I intentionally made it so the scrollbar is all the way to the right in the dialog. It looks strange otherwise. :)

**Comment by MorrisJobke** [comment]
> @MorrisJobke I intentionally made it so the scrollbar is all the way to the right in the dialog. It looks strange otherwise. :)

So you think this looks good? 😉 

bildschirmfoto 2016-10-20 um 14 38 45

**Comment by MorrisJobke** [comment]
My version:

bildschirmfoto 2016-10-20 um 14 39 57

Your version:

bildschirmfoto 2016-10-20 um 14 40 20


I still like the one with the consistent paddings 😉 Any other opinions? cc @nextcloud/designers

**Comment by jancborchardt** [comment]
@MorrisJobke right … also the scrollbar has a white background. And it would need to be flush on the left then too.

:+1: for now, any improvements can come later. Thanks!

**Comment by skjnldsv** [comment]
Agreed with @MorrisJobke. Seems too far to the right :)

**Comment by Espina2** [comment]
I think the position of the button is very weird. And I think the shadow you use in all over the place is to strong and spread. If this are a popup the background should be more black to.

**Pull request merged**